### DISCUSSION TITLE
creation of NACHA files

### DISCUSSION THREAD

**Discussion by josharmenta** [ledgersmb/LedgerSMB#6688] [Ideas]
It'd be amazing if I could generate NACHA files for ACH transactions like I can print checks. Thanks!

**Comment by ehuelsmann** [comment]
Hi @josharmenta ,

Thanks for your request. Indeed it would be great to be able to allow creation of payment files (NACHA for US or PAIN/iso20022 for EU/others).

I found documentation of the format here: https://dev-ach-guide.pantheonsite.io/ , but the number of fields specified is quite daunting. We could use a bit of help identifying the fields that need to be available from LedgerSMB to correctly fill the file. Would you be able to help us with that mapping?

Thanks!

**Comment by ehuelsmann** [comment]
For now, this issue isn't actionable (needs more discussion and analysis). Moving it to discussions for that purpose.

**Comment by josharmenta** [comment]
Yes, sure I can help. I'm happy to support in whatever way I can; I just don't know perl well enough to contribute; I'm more of a C++/Lisp guy. There's a library for Node I know of that can do the generation, is there an FFI from Perl to JS? That might be a quick way to solve the problem.

**Reply by ehuelsmann** [reply]
@josharmenta actually, the help that I need is example data, e.g. what the file would look like for your company.

If you still want this, I'm available to build it with priority for a GitHub sponsor.
